components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/RequestFilter.java

```diff
@@ -32,31 +34,31 @@
  */
 public class RequestFilter {
 
-    public static Map<String, String[]> getFilteredHeaders(Map<String, String[]> headers, ActionType actionType) {
+    public static List<Header> getFilteredHeaders(List<Header> headers, ActionType actionType) {
 
-        Map<String, String[]> filteredHeaders = new HashMap<>();
+        List<Header> filteredHeaders = new ArrayList<>();
         Set<String> excludedHeaders = ActionExecutorConfig.getInstance()
                 .getExcludedHeadersInActionRequestForActionType(actionType);
 
-        headers.forEach((key, value) -> {
+        headers.forEach(header -> {
             // Headers are case-insensitive. Therefore, convert the key to lowercase before comparing.
-            if (!excludedHeaders.contains(key.toLowerCase(Locale.ROOT))) {
-                filteredHeaders.put(key, value);
+            if (!excludedHeaders.contains(header.getName().toLowerCase(Locale.ROOT))) {
+                filteredHeaders.add(header);
             }
```

Comment:
Given that there's a new model for this we can move lower casing header name to the setter it self. So that whenever header is retrieved it's the lower case header and we don't need to give this overhead to the consumer of the header

Reply:
This is updated with 311b71367e43d41972298acf743b6eb05824ff58